tiledb/sm/query/readers/sparse_global_order_reader.cc

```diff
@@ -855,15 +919,37 @@ SparseGlobalOrderReader<BitmapType>::merge_result_cell_slabs(
     uint64_t num_cells, CompType cmp) {
   auto timer_se = stats_->start_timer("merge_result_cell_slabs");
   std::vector<ResultCellSlab> result_cell_slabs;
+  auto cmp_no_timestamps = cmp;
 
   // TODO Parallelize.
 
   // For easy reference.
-  auto dups = array_schema_.allows_dups() || consolidation_with_timestamps_;
+  const bool dups =
+      array_schema_.allows_dups() || consolidation_with_timestamps_;
+
+  // Are we doing purge deletes consolidation. The consolidation with
+  // timestamps flag will be set and we will have a post query condition
+  // bitmap. The later is only true in consolidation when delete conditions
+  // are present.
+  const bool purge_deletes_consolidation = !deletes_consolidation_no_purge_ &&
```

Comment:
IMO both of these new variables could be helper functions. The conditionals below are really hairy, so anything we can do to shorten this function would help readability.

Reply:
(since we can't modify the reader API, could be local helpers in anonymous namespace)